### PR TITLE
Dockerfile: add golang tag

### LINKED ISSUE #95: Dockerfile Latest Version
I would never use the latest tag in a docker file, because it is unstable.
You will never now with which version you end up. For example ubuntu:latest is for now ubuntu:focal.
I would prefer to use explicit versions.

Same for the golang container. Suddenly the container updated and headscale maybe won't build anymore.

Good blog about why latest is not so great:
https://vsupalov.com/docker-latest-tag/

### PR DESCRIPTION
**Pull request by ohdearaugustin**
Closes #95

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:latest AS build
+FROM golang:1.17.1-bullseye AS build
```
Comment (kradalby) [inline Dockerfile:1]:
```suggestion
FROM golang:1.17-bullseye AS build
```

How about only tagging minor so we automatically get security/patch fixes?

Comment (ohdearaugustin) [reply Dockerfile:1]: would be an idea, but we would not be able anymore to rebuild old containers as they are. Reproducible builds.

Comment (ohdearaugustin) [reply Dockerfile:1]: Maybe add something like depenendbot or something like that.

Comment (kradalby) [reply Dockerfile:1]: Dependabot or renovate would make sense regardless.


I am not sure how go and go.mod works to the detail of reproducible builds?

Comment (ohdearaugustin) [reply Dockerfile:1]: As far as I know are the dependencies  in go.mod have versions and they are not automatically updated. Also with the tags in the repo. The version of the dependency of the tag would be used.

**Comment by ohdearaugustin** [comment]
Anything on that?

**Comment by juanfont** [comment]
Apologies for the delay. Merged!

**Comment by ohdearaugustin** [comment]
Thx (=

**Pull request merged**